core/arch/arm/plat-stm32mp1/stm32_util.h

```diff
@@ -61,6 +61,9 @@ void stm32_clock_disable(unsigned long id);
 unsigned long stm32_clock_get_rate(unsigned long id);
 bool stm32_clock_is_enabled(unsigned long id);
 
+/* Return true if and only if non-secure world use the resource */
+bool stm32mp_nsec_can_access_clock(unsigned long id);
```

Comment:
Your comment and the use of "if and only if" (iff) here makes me wonder if you mean that the function would return false when both secure and non-secure can access? In which case "iff" would not be the proper term; should be "if non-secure can access and secure can not" (and the name of the function would be misleading too).
I'm asking because IMO "iff" is always implied for a boolean test function (returns true if the thing is true, and the other way around: the thing is true if it returns true).

Reply:
Then I think I should rephrase this description.
A clock can be shared by secure and non-secure. This function returns whether non-secure world is allowed to access the clock, whatever secure world can do with the clock.
Maybe the below (dropping `and only if`)?
```c
/* Return true if @clock_id is shared by secure and non-secure worlds */
bool stm32mp_nsec_can_access_clock(unsigned long clock_id);
```
as a companion to existing `stm32mp_nsec_can_access_reset()`  from https://github.com/OP-TEE/optee_os/blob/06bdcfe6e4f92fd25eb3a1542b1faa41ffca002c/core/arch/arm/plat-stm32mp1/stm32_util.h#L82-L83

Reply:
```C
/* Return true if @clock_id is shared by secure and non-secure worlds */
bool stm32mp_nsec_can_access_clock(unsigned long clock_id);
```
Perfectly fine with me :+1:
Thanks.